### DISCUSSION TITLE
What's the reasoning for registering Commands as ICommand in the registrar

### DISCUSSION THREAD

**Discussion by nils-a** [spectreconsole/spectre.console#532] [Q&A] [answered]
and what is supposed to happen?

Background: I'm trying to implement an `ITypeRegistrar` and I am wondering why every command is [registered as `ICommand`](https://github.com/spectreconsole/spectre.console/blob/f5a9c0ca26fb0d0a15a154e3e94c86797e1b1010/src/Spectre.Console/Cli/Internal/Extensions/TypeRegistrarExtensions.cs#L30) (additionally to the registration as it's own CommandType).

It seems the `ITypeResolver` does not implement a way to get all registrations of a given type but [only "one"](https://github.com/spectreconsole/spectre.console/blob/3545e0f6b5aa1c381ca14fd17b1e3c58d2fee226/src/Spectre.Console/Cli/ITypeResolver.cs#L15).

From looking at the [DefaultTypeResolver](https://github.com/spectreconsole/spectre.console/blob/a690ce49556615fea49e61972646eb52a11bbdb5/src/Spectre.Console/Cli/Internal/Composition/DefaultTypeResolver.cs#L59) I guess what is expected to happen is that the last registration for a type is used and all other registrations for that type are ignored. (Which is fine for me and probably the default for some DI systems out there.) 
But since the registration as `ICommand` is never resolved internally, is there a reason why it is registered in the first place?

**Comment by patriksvensson** [comment, marked as answer]
@nils-a Probably something that was required before, but not anymore. Could most probably be removed.
